File: backend/src/main/java/com/nexus/backend/service/CertificationService.java
package com.nexus.backend.service;

import com.nexus.backend.dto.request.AddCertificationRequest;
import com.nexus.backend.dto.request.UpdateCertificationRequest;
import com.nexus.backend.dto.response.CertificationResponse;

import java.util.List;

public interface CertificationService {

    CertificationResponse addCertification(AddCertificationRequest request);

    List<CertificationResponse> getMyCertifications();

    CertificationResponse getCertificationById(Long id);

    CertificationResponse updateCertification(Long id,
                                              UpdateCertificationRequest request);

    void deleteCertification(Long id);

}

File: frontend/src/services/certificationService.js
import api from "./api";

// ==========================================
// GET MY CERTIFICATIONS
// ==========================================

export const getMyCertifications = async () => {
    const response =
        await api.get("/certifications");

    return response.data;
};

// ==========================================
// ADD CERTIFICATION
// ==========================================

export const addCertification = async (data) => {
    const response =
        await api.post(
            "/certifications",
            data
        );

    return response.data;
};

// ==========================================
// UPDATE CERTIFICATION
// ==========================================

export const updateCertification =
    async (id, data) => {

        const response =
            await api.put(
                `/certifications/${id}`,
                data
            );

        return response.data;
    };

// ==========================================
// DELETE CERTIFICATION
// ==========================================

export const deleteCertification =
    async (id) => {

        await api.delete(
            `/certifications/${id}`
        );
    };